packages/core/src/browser/tree/tree.spec.ts

```diff
@@ -116,6 +129,83 @@ describe('Tree', () => {
 }`, node);
   });
 
+  let model: TreeModel;
+  beforeEach(() => {
+    model = createTreeModel();
+    model.root = MockTreeModel.HIERARCHICAL_MOCK_ROOT();
+  });
+  describe('getNode', () => {
+    it('returns undefined for undefined nodes', done => {
+      expect(model.getNode(undefined)).to.be.undefined;
+      done();
+    });
+
+    it('returns undefined for a non-existing id', done => {
+      expect(model.getNode('10')).to.be.undefined;
+      done();
+    });
+
+    it('returns a valid node for existing an id', done => {
+      expect(model.getNode('1.1')).not.to.be.undefined;
+      done();
+    });
+  });
+
+  describe('validateNode', () => {
+    it('returns undefined for undefined nodes', done => {
+      expect(model.validateNode(undefined)).to.be.undefined;
+      done();
+    });
+
+    it('returns undefined for non-existing nodes', done => {
+      expect(model.validateNode(MockTreeModel.Node.toTreeNode({'id' : '10'}))).to.be.undefined;
+      done();
+    });
+
+    it('returns a valid node for an existing node', done => {
+      expect(model.validateNode(retrieveNode<TreeNode>('1.1'))).not.to.be.undefined;
+      done();
+    });
+  });
+
+  describe('refresh', () => {
+    it('refreshes all composite nodes starting with the root', done => {
+      let result: Boolean = true;
+      const expectedRefreshedNodes = new Set([
+        retrieveNode<CompositeTreeNode>('1'),
+        retrieveNode<CompositeTreeNode>('1.1'),
+        retrieveNode<CompositeTreeNode>('1.2')]);
+      model.onNodeRefreshed((e: Readonly<CompositeTreeNode>) => {
+        result = result && expectedRefreshedNodes.has(e);
+        expectedRefreshedNodes.delete(e);
+      });
+      model.refresh().then(() => {
+        expect(result).to.be.true;
```

Comment:
The tests seem to fail because this result is expected to be `false` apparently.
When I made the change locally and re-ran the tests they all seemed to pass.

Reply:
I was looking into as to why this may be the case, and I believe it is because of the implementation of `refresh()`. It specifies that only the `children` nodes are refreshed, therefore the root isn't.

https://github.com/theia-ide/theia/blob/e35ba8830c9deb07318c1f6929b2cd81c01b347b/packages/core/src/browser/tree/tree.ts#L44-L47

Reply:
when executing the tests locally, then the root is getting refreshed as well